registry-manager-service-impl/src/main/java/de/adorsys/registry/manager/service/impl/AspspCsvServiceImpl.java

```diff
@@ -64,8 +67,68 @@ public AspspCsvServiceImpl(AspspRepository aspspRepository, AspspCsvRecordConver
     @Override
     public void importCsv(byte[] file) {
         List<AspspPO> aspsps = readAllRecords(file);
-        aspspRepository.deleteAll();
-        aspspRepository.saveAll(aspsps);
+        aspspRepository.delete();
+        aspspRepository.saveAll(uuidGeneratorService.checkAndUpdateUUID(aspsps));
+    }
+
+    @Override
+    public void deserializeAndMerge(byte[] file) {
+        List<AspspPO> input = readAllRecords(file);
+        List<AspspPO> database = aspspRepository.findAll();
+        List<AspspPO> forDeleting = new LinkedList<>(), forSave = new LinkedList<>();
+
+        database.forEach(item -> {
+            input.forEach(element -> {
+                if (element.getId() == null) {
+                    logicForProcessingWithNullId(input, forSave, item, element);
+//                no match by id, but match by BIC and BLZ
+                } else if (areBicAndBlzEqualWithDifferentId(item, element)) {
+                    forDeleting.add(item);
+                }
+
+//                no match, match by id, or id is NULL and no match by BIC and BLZ
+                forSave.add(element);
+            });
+        });
+
+//        everything that has been matched by BIC and BLZ but not by id (id is NOT NULL)
+        aspspRepository.delete(forDeleting);
+
+//        everything that has been matched by id, no match by id (new entries), id is NULL and matched by BIC and BLZ and no match with NULL id (new entries)
+        aspspRepository.saveAll(uuidGeneratorService.checkAndUpdateUUID(forSave));
+    }
+
+    private void logicForProcessingWithNullId(List<AspspPO> input, List<AspspPO> forSave, AspspPO item, AspspPO element) {
+//        input id is NULL, but match by BIC and BLZ
+        if (areBicAndBlzEqual(item, element)) {
+            AspspPO copy = copyContent(element);
+            copy.setId(item.getId());
+            forSave.add(copy);
+            input.remove(element);
+        }
+    }
+
+    private boolean areBicAndBlzEqual(AspspPO item, AspspPO element) {
+        return element.getBic().equals(item.getBic()) && element.getBankCode().equals(item.getBankCode());
```

Comment:
BIC should be compared without case sensitive

Reply:
done